### PR TITLE
Modify update_clusterInfo function

### LINKED ISSUE #254: The filtered cluster browsing could be improved
The cluster should be ordered by size (area)

### PR DESCRIPTION
**Pull request by Bachibouzouk**
Fix #254

### CODE REVIEW

File: app/static/js/sidebar.js [hunk 1]
```diff
@@ -1125,33 +1087,39 @@ function filter_centroid_keys(){
...
+  // sort the keys according to area
+  filtered_centroids_keys.sort(function(a, b) {
+      return a.area < b.area;
+  });
+  // only keep the sorted keys
+  var answer = [];
+  for (var i = 0; i<filtered_centroids_keys.length; i++) {
+      answer[i] = filtered_centroids_keys[i].key;
+  }
+  return answer;
```
Comment (Bachibouzouk) [inline app/static/js/sidebar.js:1120]: This part of the code sort the keys with respect to largest area order

File: app/static/js/sidebar.js [hunk 2]
```diff
@@ -1054,30 +1026,20 @@ function update_centroids(){
             centroidsGroup.addLayer(centroids_layer);
             // store the _leaflet_id of the centroids layer in a variable. The layer can be called with this id.
             centroids_layer_id = centroidsGroup.getLayerId(centroids_layer)
-            // TODO duplicate this for all_clusters as well
             // store this id in a dict with state name as keys
             centroids_layer_ids[centroids_file_key][cluster_type] = centroids_layer_id
+            //update the filters
+            update_filter()
```
Comment (Bachibouzouk) [inline app/static/js/sidebar.js:1036]: Has to be done that way because of the asynchronous call

File: app/static/js/sidebar.js [hunk 3]
```diff
@@ -309,55 +318,38 @@ noUiSlider.create(ogBuildingsFootprintSlider, {
...
+        }
+        var new_text = "= " + num_filtered_clusters + " settlements";
+        if (num_filtered_clusters == 1){
+            new_text = "= " + num_filtered_clusters + " settlement";
+        };
+        filter_title.text(new_text);
+        filter_title = $("#filtered-clusters-num");
+        filter_title.text(num_filtered_clusters);
     }
+    return num_filtered_clusters;
```
Comment (Bachibouzouk) [inline app/static/js/sidebar.js:349]: Combine og an all filters in one and get the info from the current geojson layer

File: app/static/js/sidebar.js [hunk 4]
```diff
@@ -614,25 +605,16 @@ function state_button_fun(trigger="button") {
   // manages the layers for the state level
   adapt_view_to_state_level();
 
+  update_centroids("state_button_615");
```
Comment (Bachibouzouk) [inline app/static/js/sidebar.js:608]: Useful to debug when the same update function is called at multiple points

File: app/static/js/sidebar.js [hunk 5]
```diff
@@ -1125,33 +1087,39 @@ function filter_centroid_keys(){
         centroids[key].feature.properties.percentage_building_area > currentfilter.ogminbfp && 
         centroids[key].feature.properties.percentage_building_area < currentfilter.ogmaxbfp
       ){
-        filtered_centroids_keys.push(key);
+        filtered_centroids_keys.push({"key": key, "area": centroids[key].feature.properties.area_km2});
         og_centroids_dict[centroids[key].feature.properties.cluster_offgrid_id] = key;
       }
     }
     else if (centroids[key].feature.properties.hasOwnProperty('cluster_all_id')){
-      console.log("CLUSTERS 4 ALLLLLLLLLL");
```
Comment (christian-rli) [inline app/static/js/sidebar.js:1133]: reasonable decision

**Review by christian-rli — APPROVED** [review]
It looks good, and works, but let's talk about it again, if you have time later today. I think we might be following two approaches that do a similar thing in parallel.

**Comment by christian-rli** [comment]
Also, I agree with how the legend is updated :) but the clusters don't seem to be sorted by largest area for me. Why would that be?

**Comment by christian-rli** [comment]
FYI @Bachibouzouk if cluster numbers don't add up, I just need to update the tileserver again. Planned to do this in the evening.

**Comment by christian-rli** [comment]
> Also, I agree with how the legend is updated :) but the clusters don't seem to be sorted by largest area for me. Why would that be?

My fault. I tested the wromg branch :) All good!

**Pull request merged**